control-plane-operator/controllers/hostedcontrolplane/oauth/params.go

```diff
@@ -152,6 +153,15 @@ func NewOAuthServerParams(hcp *hyperv1.HostedControlPlane, globalConfig globalco
 			p.LoginURLOverride = annotationValue
 		}
 	}
+
+	if explicitNonRootSecurityContext {
```

Comment:
Instead of repeating this code over and over again, can we please just add a `SetSecurityContext(explicitNonRootSecurityContext bool)` func to the `DeploymentConfig` type that does this?